### PR TITLE
Use a counter instead of array in lengthOfInt

### PR DESCRIPTION
**Pull request by otaviolima**

### CODE REVIEW

File: Foundation/ByteCountFormatter.swift [hunk 1]
```diff
@@ -354,14 +354,13 @@ open class ByteCountFormatter : Formatter {
     // A helper method to return the length of an int
     private func lengthOfInt(number: Int) -> Int {
         var num = abs(number)
-        var length: [Int] = []
+        var length = 0
```
Comment (spevans) [inline Foundation/ByteCountFormatter.swift]: I see this code is only used to test for length of == 3 or > 3 but maybe length should be at least 1 for a value of 0?

Comment (otaviolima) [reply Foundation/ByteCountFormatter.swift]: That is true, should I fix it in same PR or open a new one?

Comment (parkera) [reply Foundation/ByteCountFormatter.swift]: I think this one is small enough that we can put that fix here as well.

Comment (otaviolima) [reply Foundation/ByteCountFormatter.swift]: Done.

**Comment by parkera** [comment]
@swift-ci test and merge

**Comment by spevans** [comment]
@swift-ci test and merge

**Pull request merged**